publish.sh

```diff
@@ -2,8 +2,34 @@
 
 set -euo pipefail
 
+cd "$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+
+cmdexists() {
+  command -v "$1" >/dev/null 2>&1
+}
+
+if ! cmdexists gsed ; then
+  if cmdexists sed && sed --version | grep -q '(GNU sed)' ; then
+    # gsed doesnt exist on many distros; instead, sed 
+    gsed() {
+      sed "$@"
+    }
+  else
+    echo 'FATAL ERROR: GNU sed is needed by this script but is not installed. Please install gsed' 1>&2
+    exit 1
+  fi
+fi
+
+if cmdexists apt-get && cmdexists sudo ; then
+  # according to order of operations, this is ! ( cmdexists aws && cmdexists git )
+  if ! cmdexists aws && cmdexists git ; then
+    echo 'Need to install aws and git in order to proceed' 1>&2
+    sudo apt-get install -y awscli git
```

Comment:
I'm not a fan of installing stuff automatically here; this is an administrative script that's only ever run by maintainers so I don't think we need this check here -- if you have write permission to the production infrastructure, I think we can assume you know what to do if you get the error `aws: no such command`.

Definitely appreciate the cross-platform support and the better technique for getting `OLDSHA`!

Reply:
Understood.